contracts/core/Dispatcher.sol

```diff
@@ -408,24 +392,11 @@ contract Dispatcher is Ownable2StepUpgradeable, UUPSUpgradeable, ReentrancyGuard
             // _portChannelMap is used to check ownership of a channel by a port
             revert IBCErrors.channelNotOwnedBySender();
         }
-        (bool success, bytes memory data) = _callIfContract(
-            msg.sender,
-            abi.encodeWithSelector(
-                IbcChannelReceiver.onChanCloseInit.selector,
-                channelId,
-                channel.counterpartyPortId,
-                channel.counterpartyChannelId
-            )
-        );
 
```

Comment:
nit: can we refactor channel ownership check into a common method? The check is done in multiple places.  Now the full channel is no longer required after `onChanCloseInit` is removed.

Reply:
ah good one. This check is done in 3 different places - sendpacket, channelCloseInit, and channelCloseConfirm. 

Channel close confirm requires the channel afterwards, so to avoid adding an extra SLOAD in channelCloseConfirm or forcing sendpacket & channelCloseInit to call a method which returns more data than it needs to (increasing gas cost considerably), I didn't have this use the common method.

We can use the same method for sendpacket and channelCloseInit though! FYI, this does seem to add an extra 40 bytes in dispatcher contract size (it seems like the code for the method itself is larger than encoding/decoding the data into an internal function), but this is _probably_ negligible since we're now working with 1600 bytes of room in our dispatcher.